security/kms-key.yaml

```diff
@@ -55,18 +57,44 @@ Parameters:
     - sqs
     - workmail
     - workspaces
+    - dnssec-route53
     Default: ALL_SERVICES
+  KeySpec:
+    Description: 'Specify the type of the CMK.'
+    Type: String
+    AllowedValues:
+    - SYMMETRIC_DEFAULT
+    - RSA_2048
+    - RSA_3072
+    - RSA_4096
+    - ECC_NIST_P256
+    - ECC_NIST_P384
+    - ECC_NIST_P521
+    - ECC_SECG_P256K1
+    Default: SYMMETRIC_DEFAULT
+  KeyUsage:
+    Description: 'Which cryptographic operations should the CMK support?'
+    Type: String
+    AllowedValues:
+    - ENCRYPT_DECRYPT
+    - SIGN_VERIFY
+    Default: ENCRYPT_DECRYPT
 Conditions:
   HasAlertTopic: !Not [!Equals [!Ref ParentAlertStack, '']]
   HasServiceAllServices: !Equals [!Ref Service, 'ALL_SERVICES']
   HasServiceS3PublicAccess: !Equals [!Ref Service, 'S3_PUBLIC_ACCESS']
+  HasDNSSECRoute53Access: !Equals [!Ref Service, 'dnssec-route53']
+  HasNotDNSSECRoute53Access: !Not [!Equals [!Ref Service, 'dnssec-route53']]
+  HasSymmetricKey: !Equals [!Ref KeySpec, 'SYMMETRIC_DEFAULT']
 Resources:
   Key:
     DeletionPolicy: Retain
     UpdateReplacePolicy: Retain
     Type: 'AWS::KMS::Key'
     Properties:
-      EnableKeyRotation: true
+      EnableKeyRotation: !If [HasSymmetricKey, true, false]
+      KeySpec: !Ref KeySpec
```

Comment:
have you tested if this will replace an existing key if the defaults are used?

Reply:
Good point. Tested. Does not replace existing keys.